ui/hooks/useTransactionDisplayData.js

```diff
@@ -37,17 +37,41 @@ import { useTokenData } from './useTokenData';
 import { useSwappedTokenValue } from './useSwappedTokenValue';
 import { useCurrentAsset } from './useCurrentAsset';
 
+/**
+ *  There are seven types of transaction entries that are currently differentiated in the design:
+ *  1. Signature request
+ *  2. Send (sendEth sendTokens)
+ *  3. Deposit
+ *  4. Site interaction
+ *  5. Approval
+ *  6. Swap
+ *  7. Swap Approval
+ */
+const signatureTypes = [
+  null,
+  undefined,
+  TRANSACTION_TYPES.SIGN,
+  TRANSACTION_TYPES.PERSONAL_SIGN,
+  TRANSACTION_TYPES.SIGN_TYPED_DATA,
+  TRANSACTION_TYPES.ETH_DECRYPT,
+  TRANSACTION_TYPES.ETH_GET_ENCRYPTION_PUBLIC_KEY,
+];
+
+/**
+ * @typedef {(import('../../selectors/transactions').TransactionGroup} TransactionGroup
+ */
+
 /**
  * @typedef {Object} TransactionDisplayData
- * @property {string} title - primary description of the transaction
- * @property {string} subtitle - supporting text describing the transaction
- * @property {bool} subtitleContainsOrigin - true if the subtitle includes the origin of the tx
- * @property {string} category - the transaction category
+ * @property {string} category - the transaction category that will be used for rendering the icon in the activity list
  * @property {string} primaryCurrency - the currency string to display in the primary position
- * @property {string} [secondaryCurrency] - the currency string to display in the secondary position
- * @property {string} status - the status of the transaction
- * @property {string} senderAddress - the Ethereum address of the sender
  * @property {string} recipientAddress - the Ethereum address of the recipient
+ * @property {string} senderAddress - the Ethereum address of the sender
+ * @property {string} status - the status of the transaction
+ * @property {string} subtitle - the supporting text describing the transaction
+ * @property {boolean} subtitleContainsOrigin - true if the subtitle includes the origin of the tx
+ * @property {string} title - the primary title of the tx that will be displayed in the activity list
+ * @property {string} [secondaryCurrency] - the currency string to display in the secondary position
```

Comment:
this is kind of out of scope of this PR but noting that using an object first naming convention, putting adjectives further in the name, would solve the only issue I take with this PR: breaking apart obviously related things like primaryCurrency and secondaryCurrency. Not blocking, just noting, we may want to officially adopt a object first naming convention. 

`currencyPrimary` and `currencySecondary` would probably help with readability all over.

Reply:
Thank you for noting this and I totally agree! I'm on board for object first naming convention.

I have proposed this in a comment in our internal [Code Guidelines (Draft) ](https://www.notion.so/Code-Guidelines-DRAFT-02a7e4d7d62b4c9eaab8d4d0669e0d75?d=7fa22e7f90234294934c2704f5d25b63#562d731a9d9e44efabe55db29fb99250)